packages/apps/desktop-viewer-test/src/frontend/components/modelSelector/SelectIModel.tsx

```diff
@@ -174,27 +187,39 @@ export const SelectIModel = ({
   const userSettings = useContext(SettingsContext);
   const modelContext = useContext(IModelContext);
 
-  const selectIModel = async (iModel: IModelFull) => {
-    if (modelContext.pendingIModel) {
-      // there is already a pending selection. disallow
-      return;
-    }
-    const recents = userSettings.settings.recents;
-    if (recents) {
-      const local = recents.find((recent) => {
-        return (
-          recent.iTwinId === iModel.iTwinId && recent.iModelId === iModel.id
-        );
-      });
-      if (local?.path) {
-        // already downloaded, navigate
-        void navigate("/viewer", { state: { filePath: local.path } });
+  const selectIModel = useCallback(
+    async (iModel: IModelFull) => {
+      if (modelContext.pendingIModel) {
+        // there is already a pending selection. disallow
         return;
       }
-    }
-    // trigger a download/view
-    modelContext.setPendingIModel(iModel.id);
-  };
+
+      const recents = userSettings.settings.recents;
```

Comment:
Similar code is repeated in `getLocal` above. Perhaps some of this could be extracted into a common function.